base/broadcast.jl

```diff
@@ -623,8 +623,44 @@ eltypes(t::Tuple{Any}) = Tuple{_broadcast_getindex_eltype(t[1])}
 eltypes(t::Tuple{Any,Any}) = Tuple{_broadcast_getindex_eltype(t[1]), _broadcast_getindex_eltype(t[2])}
 eltypes(t::Tuple) = Tuple{_broadcast_getindex_eltype(t[1]), eltypes(tail(t)).types...}
 
+function promote_typejoin_union(::Type{T}) where T
+    if T === Union{}
+        return Union{}
+    elseif T isa Union
+        promote_typejoin(promote_typejoin_union(T.a), promote_typejoin_union(T.b))
+    elseif T <: Tuple
+        typejoin_union_tuple(T)
+    else
+        T
+    end
+end
+
+@pure function typejoin_union_tuple(::Type{T}) where {T}
+    p = T.parameters
+    lr = length(p)::Int
+    if lr == 0
+        return Tuple{}
+    end
+    lf, fixed = Core.Compiler.full_va_len(p)
+    c = Vector{Any}(undef, lf)
+    for i = 1:lf
+        pi = p[i]
+        U = Core.Compiler.unwrapva(pi)
+        if U === Union{}
+            ci = Union{}
+        elseif U isa Union
+            ci = typejoin(U.a, U.b)
+        else
+            ci = U
+        end
+        c[i] = i == lf && Core.Compiler.isvarargtype(pi) ? Vararg{ci} : ci
```

Comment:
```suggestion
        c[i] = i == lf && Core.Compiler.isvarargtype(pi) ? Vararg{ci, N} : ci
```

The Vararg parameter needs to be extracted from `pi` also